python/tk_multi_data_validation/dialog.py

```diff
@@ -243,10 +243,30 @@ def _resolve_rules(self, rules):
 
         try:
             self._manager.resolve_rules(rules)
+            self._bundle.execute_hook_method(
+                "hook_data_validation",
+                "post_fix_action",
+                rule_ids=[r.id for r in rules],
+            )
```

Comment:
As suggested above, let's move this whole `_resolve_rules` function to a hook. When migrating it over, we can clean this function up by removing the show/hide busy pop up calls (this is not needed and redundant because the show/hide busy pop ups are shown also by signal/slots for `resolve_all_begin` and `resolve_all_finished`. So your hook function to replace this would look something like:

```
def resolve_rules(self, manager, rules):
    manager.resolve_rules(rules)
    self.post_fix_action(rules)
    manager.validate_rules(rules)
```